### ISSUE TITLE
ssh autocomplete for fish shell no longer works

### ISSUE THREAD

**Issue by ElijahLynn** [warpdotdev/Warp#4961]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

Recently SSH autocompletion with my fish shell worked fine, something broke it in the past week or so. This is mainly a rubber ducking thread and I'll post my results here. This might be a fish issue, might be a warp issue. But I do know that if I open a non-Warp shell then SSH autocompletion works as expected. 

### To reproduce

type `ssh` then `<TAB>`. A list of IP addresses appear, but no hostnames. It appears that it does list suggestions IF the hostname is in ~/.ssh/config but NOT any `Include <path>/*` directives. 

### Expected behavior

List of hosts in ~/.ssh/config to return. 

### Operating system

MacOS

### Operating system and version

14.5 (23F79)

### Shell Version

fish, version 3.7.1

### Current Warp version

v0.2024.05.14.08.01.stable_04

### Regression

Yes, this bug started recently or with an X Warp version

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Warp has an SSH Wrapper that enables features (like the blocks, prompt, completions, etc.) on remote machines requiring the default prompt on the remote machine to be either `bash` or `zsh`.

As a workaround you can add `command ssh` to your `Settings > Subshells > Added commands` menu, then run `command ssh <user@server>` to connect to remote session, this will attempt to enable Warp features as a subshell.

You can also try disabling the SSH Wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabling, SSH should work as normal.

Related docs:
[https://docs.warp.dev/features/ssh](https://docs.warp.dev/features/ssh#ssh-wrapper-fails)
https://docs.warp.dev/features/subshells

If those don't work, please let us know.

**Comment by ElijahLynn** [comment]
Thanks, I should have noted that I have already disabled the SSH wrapper when experiencing this issue.

**Comment by dannyneira** [comment]
yes the subshell method is a workaround to use `fish` shell on remote machines. Let me know if that helped

**Comment by ElijahLynn** [comment]
Gosh, I forgot I opened this and just opened an issue on Fish, then realized it was Warp that was the cause of the issue:
* https://github.com/fish-shell/fish-shell/issues/10632

The issues appears to only affect `Include <path>/*` directives inside ~/.ssh/config. Those hostnames that are Included do not appear in completion. Where is the source code that Warp uses to Include? I don't see it in this repo.
